src/gssapi_mechanism_base.cpp

```diff
@@ -181,12 +181,18 @@ int zmq::gssapi_mechanism_base_t::decode_message (msg_t *msg_)
     maj_stat = gss_unwrap(&min_stat, context, &wrapped, &plaintext,
                           &state, (gss_qop_t *) NULL);
 
-    //  TODO I don't think it is a good idea to use zmq_assert here. If 
-    //  decryption fails, gss_unwrap returns GSS_S_BAD_SIG. This opens up 
-    //  to DoS attacks by clients! Instead, a 
-    //  ZMQ_PROTOCOL_ERROR_ZMTP_CRYPTOGRAPHIC event should be emitted.
-
-    zmq_assert(maj_stat == GSS_S_COMPLETE);
+    if (maj_stat != GSS_S_COMPLETE)
+    {
+        //  TODO is it correct to release the plaintext buffer if gss_unwrap 
+        //  did not succeed?
+        gss_release_buffer (&min_stat, &plaintext);
```

Comment:
Based on the documentation of gss_unwrap, I could not tell.
But I found several code examples that also do call gss_release_buffer in an error case.
For example: https://github.com/happyhappiness/vscode/blob/650a7dbf7303fa37e5096b1ef667d2e7aae04872/second/download/CMake/CMake-old-new/CMake-old-new/Kitware_CMake_old_file_98.cpp#L482